WorkspaceServer.Tests/Kernel/CSharpKernelTests.cs

```diff
@@ -407,59 +409,43 @@ public async Task The_extend_directive_can_be_used_to_load_a_kernel_extension()
         {
             var extensionDir = Create.EmptyWorkspace()
                                      .Directory;
+            var outputDir = extensionDir.CreateSubdirectory("outputDir");
 
             var microsoftDotNetInteractiveDllPath = typeof(IKernelExtension).Assembly.Location;
 
-            var dirAccessor = new InMemoryDirectoryAccessor(extensionDir)
-                {
-                    ( "Extension.cs", $@"
-using System;
-using System.Reflection;
-using System.Threading.Tasks;
-using Microsoft.DotNet.Interactive;
-using Microsoft.DotNet.Interactive.Commands;
-
-public class TestKernelExtension : IKernelExtension
-{{
-    public async Task OnLoadAsync(IKernel kernel)
-    {{
-        await kernel.SendAsync(new SubmitCode(""using System.Reflection;""));
-    }}
-}}
-" ),
-                    ("TestExtension.csproj", $@"
-<Project Sdk=""Microsoft.NET.Sdk"">
-
-  <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
-  </PropertyGroup>
-
-    <ItemGroup>
-
-    <Reference Include=""Microsoft.DotNet.Interactive"">
-      <HintPath>{microsoftDotNetInteractiveDllPath}</HintPath>
-    </Reference>
-  </ItemGroup>
-
-</Project>
-")
-                }
-                .CreateFiles();
-
-            var buildResult = await new Dotnet(extensionDir).Build();
-            buildResult.ThrowOnFailure();
-
-            var extensionDllPath = extensionDir
-                                   .GetDirectories("bin", SearchOption.AllDirectories)
-                                   .Single()
-                                   .GetFiles("TestExtension.dll", SearchOption.AllDirectories)
-                                   .Single()
-                                   .FullName;
+            var extensionDllPath = (await KernelExtensionTestHelper.CreateExtension(extensionDir)).FullName;
 
             var kernel = CreateKernel();
 
             await kernel.SendAsync(new SubmitCode($"#extend \"{extensionDllPath}\""));
 
+            KernelEvents.Should().ContainSingle(e => e.Value is ExtensionLoaded &&
+                                                     e.Value.As<ExtensionLoaded>().ExtensionPath.FullName.CompareTo(extensionDllPath) == 0);
+            KernelEvents.Should()
+                        .ContainSingle(e => e.Value is CodeSubmissionEvaluated &&
+                                            e.Value.As<CodeSubmissionEvaluated>().Code.Contains("using System.Reflection;"));
+        }
+
+
+        [Fact]
+        public async Task Should_load_extension_in_nuget_package()
+        {
+            var directory = Create.EmptyWorkspace().Directory;
+
+            const string nugetPackageName = "myNugetPackage";
+            var nugetPackageDirectory = new FileSystemDirectoryAccessor(directory.Subdirectory($"{nugetPackageName}/2.0.0"));
```

Comment:
Can we test this using `InMemoryDirectoryAccessor`?

Reply:
This is an end to end test which tests that given a nuget package dll, is it able to load the extension from that, and since "LoadAssembly" is being called, it needs the artifacts to actually exist on disk.